### PR TITLE
Release notes for Percona Server 5.7.10-2rc2

### PR DESCRIPTION
**Pull request by hrvojem**

### CODE REVIEW

File: doc/source/release-notes/Percona-Server-5.7.10-2.rst [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+Known Issues
+============
+
+ In |Percona Server| 5.7 `super_read_only <https://www.percona.com/doc/percona-server/5.6/management/super_read_only.html>`_ feature has been replaced with upstream implementation. There are currently two know issues compared to |Percona Server| 5.6 implementation:
+
+   * Bug :mysqlbug:`78963`, :variable:`super_read_only` aborts ``STOP SLAVE`` if variable :variable:`relay_log_info_repository` is set to ``TABLE`` which could lead to a server crash in Debug builds.
+
+   * Bug :mysqlbug:`79328`, :variable:`super_read_only` set as a server option has no effect.
+
+ Restarting the server with the :variable:`innodb_flush_method` set to ``ALL_O_DIRECT`` fails intermittently which could break crash recovery (bug :bug:`1529885`).
```
Comment (laurynas-biveinis) [inline doc/source/release-notes/Percona-Server-5.7.10-2.rst]: InnoDB crash recovery might fail if innodb_flush_method is set to ALL_O_DIRECT. The workaround is to set this variable to a different value before starting up the crashed instance.

File: doc/source/release-notes/Percona-Server-5.7.10-2.rst [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+
+Bugs Fixed
+==========
+
+ Clustering secondary index could not be created on a partitioned |TokuDB| table. Bug fixed :bug:`1527730` (:tokubug:`720`).
+
+ :ref:`toku_backup` was failing to compile with |Percona Server| 5.7. Bug fixed :backupbug:`123`.
+
+ When enabled, :ref:`super-read-only` could break statement-based replication while executing a multi-table update statement on a slave. Bug fixed :bug:`1441259`.
```
Comment (laurynas-biveinis) [inline doc/source/release-notes/Percona-Server-5.7.10-2.rst]: This bug did not exist on 5.7, I have updated the milestone, please remove this entry

File: doc/source/release-notes/Percona-Server-5.7.10-2.rst [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+
+ Granting privileges to user with :ref:`pam_plugin` could lead to a server crash. Bug fixed :bug:`1521474`.
```
Comment (laurynas-biveinis) [inline doc/source/release-notes/Percona-Server-5.7.10-2.rst]: ...to a user authenticating with pam_plugin...

File: doc/source/release-notes/Percona-Server-5.7.10-2.rst [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+
+ |TokuDB| status variables were missing from |Percona Server| :rn:`5.7.10-1`. Bug fixed :bug:`1527364` (:tokubug:`923`).
+
+ Attempting to rotate the audit log file would result in audit log file name :file:`foo.log.%u` (literally) instead of a numeric suffix. Bug fixed :bug:`1528603`.
+
+ Adding an index to a temporary table while :variable:`expand_fast_index_creation` was enabled could lead to server assertion. Bug fixed :bug:`1529555`.
```
Comment (laurynas-biveinis) [inline doc/source/release-notes/Percona-Server-5.7.10-2.rst]: ... to an InnoDB temporary table ...

File: doc/source/release-notes/Percona-Server-5.7.10-2.rst [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+
+ |TokuDB| would not be upgraded on *Debian*/*Ubuntu* distributions while performing an upgrade from |Percona Server| 5.6 to |Percona Server| 5.7 even if explicitly requested. Bug fixed :bug:`1533580`.
+
+ Server would assert when two different transactional engines were used within one transaction on a replication slave which has binary log enabled and disabled slave updates logging (:variable:`log_slave_update` set to ``0``). Bug fixed :bug:`1534249` (upstream bug :mysqlbug:`80053`).
```
Comment (laurynas-biveinis) [inline doc/source/release-notes/Percona-Server-5.7.10-2.rst]: Server would assert when both TokuDB and InnoDB tables were used within one transaction ...
... binary log enabled and slave updates logging disabled

File: doc/source/release-notes/Percona-Server-5.7.10-2.rst [hunk 6]
```diff
@@ -0,0 +1,80 @@
...
+
+ `MeCab Full-Text Parser Plugin <https://dev.mysql.com/doc/refman/5.7/en/fulltext-search-mecab.html>`_ has not been included in the previous release. Bug fixed :bug:`1534617`.
+
+ Fixed server assertion caused by ``Performance Schema`` memory key mix-up. Bug fixed :bug:`1534874`.
```
Comment (laurynas-biveinis) [inline doc/source/release-notes/Percona-Server-5.7.10-2.rst]: in SET STATEMENT ... FOR ... statements

File: doc/source/release-notes/Percona-Server-5.7.10-2.rst [hunk 7]
```diff
@@ -0,0 +1,80 @@
...
+
+ Setting the :variable:`innodb_sched_priority_purge` (available only in debug builds) while purge threads were stopped would cause a server crash. Bug fixed :bug:`1368552`.
+
+ Enabling |TokuDB| with ``ps_tokudb_admin`` script inside the Docker container would cause an error due to insufficient privileges even when running as root. In order for this script to be used inside docker containers this error has be been changed to a warning that a check is impossible. Bug fixed :bug:`1520890`.
```
Comment (laurynas-biveinis) [inline doc/source/release-notes/Percona-Server-5.7.10-2.rst]: "has be been"

File: doc/source/release-notes/Percona-Server-5.7.10-2.rst [hunk 8]
```diff
@@ -0,0 +1,80 @@
...
+
+ `MeCab Full-Text Parser Plugin <https://dev.mysql.com/doc/refman/5.7/en/fulltext-search-mecab.html>`_ has not been included in the previous release. Bug fixed :bug:`1534617`.
+
+ Fixed server assertion caused by ``Performance Schema`` memory key mix-up. Bug fixed :bug:`1534874`.
+
+ Setting the :variable:`innodb_sched_priority_purge` (available only in debug builds) while purge threads were stopped would cause a server crash. Bug fixed :bug:`1368552`.
+
+ Enabling |TokuDB| with ``ps_tokudb_admin`` script inside the Docker container would cause an error due to insufficient privileges even when running as root. In order for this script to be used inside docker containers this error has be been changed to a warning that a check is impossible. Bug fixed :bug:`1520890`. 
+
+ Write heavy workload with small buffer pool could lead to a deadlock when free buffers are exhausted. Bug fixed :bug:`1521905`.
```
Comment (laurynas-biveinis) [inline doc/source/release-notes/Percona-Server-5.7.10-2.rst]: Write-heavy workload with a small buffer pool ...

File: doc/source/release-notes/Percona-Server-5.7.10-2.rst [hunk 9]
```diff
@@ -0,0 +1,78 @@
...
+
+ Busy server was preferring LRU flushing over flush list flushing too strongly which could lead to performance degradation. Bug fixed :bug:`1534114`.
+
+ :file:`libjemalloc.so.1` was missing from binary tarball. Bug fixed :bug:`1537129`.
+
+ When ``cmake/make/make_binary_distribution`` workflow was used to produce binary tarballs it would produce tarballs with ``mysql-...`` naming instead of ``percona-server-...``. Bug fixed :bug:`1540385`.
+
+ Bug fixed :tokubug:`917`.
+
+ Bug fixed :tokubug:`926`.
```
Comment (george-lorch) [inline doc/source/release-notes/Percona-Server-5.7.10-2.rst]: 5.7 changed the way a storage engine should report index statistics to the optimizer. This issues implements the new index statistic reporting within TokuDB. Following the InnoDB example, the default value for tokudb_cardinality_scale has changed from 50% to 100%. Implementing this addresses a server crash deep in the optimizer code that results from a combination of conditions.

File: doc/source/release-notes/Percona-Server-5.7.10-2.rst [hunk 10]
```diff
@@ -0,0 +1,78 @@
...
+
+ Page cleaner worker threads were not instrumented for ``Performance Schema``. Bug fixed :bug:`1532747` (upstream bug :mysqlbug:`79894`).
+
+ Busy server was preferring LRU flushing over flush list flushing too strongly which could lead to performance degradation. Bug fixed :bug:`1534114`.
+
+ :file:`libjemalloc.so.1` was missing from binary tarball. Bug fixed :bug:`1537129`.
+
+ When ``cmake/make/make_binary_distribution`` workflow was used to produce binary tarballs it would produce tarballs with ``mysql-...`` naming instead of ``percona-server-...``. Bug fixed :bug:`1540385`.
+
+ Bug fixed :tokubug:`917`.
```
Comment (george-lorch) [inline doc/source/release-notes/Percona-Server-5.7.10-2.rst]: If you look at the commit messate and the 5.6 pull request, you will see some more details: https://github.com/percona/percona-server/pull/329

A few things are in this fix:
- Another new tokudb_debug flag for tracking calls in/out of the 'shared' table data. This is probably not of much interest to an end user but other devs might find it useful.
- Added proper memory cleanup if for some reason a table is unable to be opened from a dead closed state. This prevents an assertion from happening the next time the table is attempted to be opened.

**Comment by laurynas-biveinis** [comment]
Non-TokuDB parts LGTM

**Pull request merged**